### ISSUE TITLE
Problems while predicting, multilabel classification

### ISSUE THREAD

**Issue by djSharma7** [ThilinaRajapakse/simpletransformers#240]
I'm trying to do multi-label classification. 
I'm having 12 labels and 7k ranging dataset as of now, i have trained the model and when I try to predict output of a single string same as of training data set, i'm getting wrong results!
**My Use Case**
E-commerce reviews sentiment analysis
I've been feeding a training set with 12 labels!
Product_-1,Product_0,Product_1,Price_-1,Price_0,Price_1,Delivery_-1,Delivery_0,Delivery_1,Packaging_-1,Packaging_0,Packaging_1
-1=>Negative
0=>Neutral
1=>Positive
```
data = pd.read_excel('train_.xlsx')
data.dropna(inplace=True)
data.reset_index(drop=True,inplace=True)
data['Delivery']=data['Delivery'].astype(int)
data['Product']=data['Product'].astype(int)
data['Price']=data['Price'].astype(int)
data['Packaging']=data['Packaging'].astype(int)
data=pd.get_dummies(data,columns=['Product','Price','Delivery','Packaging'])
data.rename(columns = {"Product_-1": "Product","Price_-1":"Price","Delivery_-1":"Delivery","Packaging_-1":"Packaging"}, inplace = True)


data['labels'] = list(zip(data.Product.tolist(), data.Product_0.tolist(), data.Product_1.tolist(),
                              data.Price.tolist(), data.Price_0.tolist(), data.Price_1.tolist(),
                                                        data.Delivery.tolist(), data.Delivery_0.tolist(), data.Delivery_1.tolist(),
                                                                                  data.Packaging.tolist(), data.Packaging_0.tolist(), data.Packaging_1.tolist()))
data['text'] = data['Review'].apply(lambda x: x.replace('\n', ' '))
from sklearn.model_selection import train_test_split
#
#
train_df, eval_df = train_test_split(data, test_size=0.2)
# Create a MultiLabelClassificationModel
if __name__ == '__main__':
    model = MultiLabelClassificationModel(
        "roberta",
        "roberta-base",
        num_labels=12,
        args={"reprocess_input_data": True, "overwrite_output_dir": True, "num_train_epochs": 5}
    )
    model.train_model(train_df)
    # Evaluate the model
    result, model_outputs, wrong_predictions = model.eval_model(eval_df)
    print(result)
    print(model_outputs)

    predictions, raw_outputs = model.predict(["good product but price is too high and delivery was 
    too late"])
    print(predictions)
    print(raw_outputs)
```

Am i doing something wrong!
Your help will be really appreciated

**Comment by ThilinaRajapakse** [comment]
What is your training loss at the end of training and what do your evaluation scores look like?

**Comment by djSharma7** [comment]
LRAP = 0.8836415172129456
eval_loss = 0.29891220331192014

**Comment by djSharma7** [comment]
Can you guide me to a right direction..
I'm bit curious if this satisfies my requirements..
As i believe it would totally satisfy if i had 4 labels only (product,price,delivery,packaging) but i have 12 labels instead (product_neg,product_neut,product_pos,price_neg,price_neut,price_pos,delivery_neg,delivery_neut,delivery_pos,packaging_neg,packaging_neut,packaging_pos)

Review example-
> Product is good, but the price is high and delivery was very late

Training set example-

> Product is good, but the price is high and delivery was very late
Product | Price, | Delivery,| Packaging
 1           |    -1   |     -1      | 0

1->Positive
-1->Negative
0->Neutral
Converting it into 12 labels-->

Prod_neg,Prod_neu,Prod_pos,Pri_neg,Pri_neu,Pri_pos,del_neg,del_neu,del_pos,pac_-,pac_neu,pac+
0                  0               1             1          0            0            1          0           0          0       1             0

I'm following this approach
Do i have any other option ! 
Help will be really appreciated..




 

>

**Comment by ThilinaRajapakse** [comment]
I think your approach is fine. A multilabel classification where each label can also have multiple values is commonly known as multi_task_ classification (see [here](https://scikit-learn.org/stable/modules/multiclass.html)). This is not supported with Simple Transformers.

I am double checking the multilabel classification task in Simple Transformers as it seems to be causing issues for several people atm. I'll let you know when I have more info.

In the meantime, are all your outputs from `predict()` also 0?

Please respond in #234 as these issues are probably related.
